gui/src/app/SPAnalysis/SPAnalysisContextProvider.tsx

```diff
@@ -12,11 +14,23 @@ type SPAnalysisContextProviderProps = {
 
 export const SPAnalysisContext = createContext<SPAnalysisContextType>({
     data: initialDataModel,
-    update: () => {}
+    update: () => { }
 })
 
-const SPAnalysisContextProvider: FunctionComponent<PropsWithChildren<SPAnalysisContextProviderProps>> = ({children}) => {
-    const [data, update] = useReducer<SPAnalysisReducerType>(SPAnalysisReducer, initialDataModel)
+
+const SPAnalysisContextProvider: FunctionComponent<PropsWithChildren<SPAnalysisContextProviderProps>> = ({ children }) => {
+    const [searchParams, setSearchParams] = useSearchParams();
+
+    const onDirty = useCallback(() => {
+        // whenever the data state is 'dirty', we want to
+        // clear the URL bar as to indiciate that the viewed content is
+        // no longer what the link would point to
+        if (searchParams.size !== 0)
+            setSearchParams(new URLSearchParams())
+    }, [searchParams, setSearchParams]);
+
+
+    const [data, update] = useReducer<SPAnalysisReducerType>(SPAnalysisReducer(onDirty), initialDataModel)
```

Comment:
I don't mind the `onDirty` pattern here; I haven't seen it before but it's a clever solution and pretty straightforward.

I would put the definition of the function in the SPAnalysisQueryLoading file though, since it's more to do with URL-munging, if it's possible to do so. (This might involve moving the `useSearchParams` hook out, or possibly just having multiple components use such a hook; it clearly has to be a singleton, so it's probably more in line with ` context` than a `state` whatever the documentation says. Or we could turn `fromSearchParams` into a hook defined in that file...)

Reply:
I turned fromSearchParams into a hook in 01ced77695b43fce0600e5f226380363a8b297ca, good suggestion